### PR TITLE
setup.md: Add instructions on choosing text editor

### LINKED ISSUE #22: Teach students how to use/configure a text editor
At the moment we are somewhat preventing students to fall into the _vim hell_ when they make a commit by suggesting the use of `--no-edit` or `-m` for a one line comment. however, at intermediate level, it might be convenient to ask them for some more proficiency on this. 

While leaving things as they are for now until we have run through the course at least one and check how the timings work, we might want to re-visit this decision and give a sort of mini-training as part of the course. 

_Originally posted by @alexdewar in https://github.com/ImperialCollegeLondon/intermediate_grad_school_git_course/pull/19#discussion_r1039273062_

### PR DESCRIPTION
**Pull request by alexdewar**
Add some instructions to `setup.md` on how to configure Git to use your preferred text editor.

While I think we should probably also show students how to make use of this (e.g. by making a commit with a longer message), that's slightly out of scope and a job for another day.

Closes #22.

### CODE REVIEW

File: setup.md [hunk 1]
```diff
@@ -115,4 +115,28 @@ output similar to the below:
 git version 2.40.0
 ```
 
+## Configure Git to use your preferred text editor
+
+Many Git commands require you to provide input via a text editor (though you can often skip this step with additional command-line arguments). For example, running `git commit` by itself will open a text editor to allow you to enter a commit message. Depending on your OS and how you installed Git, the text editor Git will use can be any number of things, but is likely to be either [nano] or [vim], which are both terminal-based editors. It is best to configure your preferred editor explicitly as vim in particular is famously challenging for beginners.
```
Comment (SaranjeetKaur) [inline setup.md:120]:
```suggestion
Many Git commands require you to provide input via a text editor (though you can often skip this step with additional command-line arguments). For example, running `git commit` by itself will open a text editor to allow you to enter a commit message. Depending on your OS and how you installed Git, the text editor Git will use can be any number of things, but is likely to be either [nano] or [vim], which are both terminal-based editors. It is best to configure your preferred editor explicitly (if you are using editor for the first time, it is advisable to use nano).
```

Comment (SaranjeetKaur) [reply setup.md:120]: Slightly rephrased the last sentence - I recall this is something I realised during [Carpentries Instructor Training](https://carpentries.github.io/instructor-training/08-motivation.html#how-can-content-influence-motivation)

**Review by SaranjeetKaur — APPROVED** [review]
Thanks @alexdewar! This is cool! 
Added a minor comment.

File: setup.md (hunk 1 above)
Comment (alexdewar) [reply setup.md]: Cool. Thanks!

**Comment by AdrianDAlessandro** [comment]
@dc2917 This change might be worth adding to your setup also

**Comment by dc2917** [comment]
Thanks for pointing this out. I had wondered about adding something along these lines when explaining the --no-edit option etc.. And yeah, the git installation instructions for windows in the setup states "“Use the nano editor by default”", so they should all have something appropriate installed.

**Pull request merged**